src/app/OperationalDeviceProxy.cpp

```diff
@@ -317,12 +317,21 @@ void OperationalDeviceProxy::OnSessionEstablished(const SessionHandle & session)
 CHIP_ERROR OperationalDeviceProxy::Disconnect()
 {
     ReturnErrorCodeIf(mState != State::SecureConnected, CHIP_ERROR_INCORRECT_STATE);
+    
     if (mSecureSession)
     {
-        mInitParams.sessionManager->ExpirePairing(mSecureSession.Get().Value());
+        //
+        // Mark the session as defunct to signal that we no longer want to use this
+        // session anymore for further interactions to this peer. However, if we receive
+        // messages back from that peer on the defunct session, it will bring it back into an active
+        // state again.
+        //
+        mSecureSession.Get().Value()->AsSecureSession()->MarkAsDefunct();
```

Comment:
If we are shutting down the stack, how do we inform the stack that we really, truly want a session to be closed?

Reply:
If it's for a specific session, you can call `MarkForEviction` on that session instead.

Note that on stack shut-down, the destruction of `SessionManager` will automatically evict and de-allocate all resident sessions in the session table automatically.

Reply:
Ah, OK.  That's great.